File: DecryptMessage.cpp
#include <iostream>
#include <string>
#include <Windows.h>
using namespace std;


string decrypting(string keys, string a){
    for( int i =0; i< a.length(); i++){
    int key = keys[i % 6] - int('A');
    
    if( int(a[i]) - key > int('A')){
    
    a[i] = char(int(a[i]) - key);
    
                                   }
     else if (int(a[i]) > int('z') || int(a[i]) < int('A')){
       continue;
                                                            }
    else {
        a[i] = char(-(int('A') - (int(a[i]) - key))+ int('z')) ;
         }
                                      }
  return a;
                                        }
int main(){
  string enc_m = "lcj'] [Yp J bYiX ]^`k] _]X hXoa hXO_.";
  string keys = "invicta";
  string dec_m = decrypting(keys, enc_m);
  cout << dec_m;
  
          }

File: DecryptMessage.py
def decrypting(keys, a){
    
    for i in range(len(a)):
        key = ord(keys[i % 6]) - ord('A')
        if(ord(a[i]) - key) > ord('A'):
            a[i] = ascii(ord(a[i]) - key)
        elif (ord(a[i] > ord('z'))):
            continue
        else:
            a[i] = ascii((-ord('A') - (ord(a[i]) - key)) + int('z'))
    return a
if __name__ == "__main__":
    enc_m = "lcj'] [Yp J bYiX ]^`k] _]X hXoa hXO_.";
    keys = "invicta"
    dec_m = decrypting(keys, enc_m)
    print(dec_m)
